hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -199,7 +201,7 @@ protected boolean isPendingCompactionScheduledForFileId(HoodieFileGroupId fgId)
   protected void resetPendingCompactionOperations(Stream<Pair<String, CompactionOperation>> operations) {
     // Build fileId to Pending Compaction Instants
     this.fgIdToPendingCompaction = createFileIdToPendingCompactionMap(operations.map(entry ->
-      Pair.of(entry.getValue().getFileGroupId(), Pair.of(entry.getKey(), entry.getValue()))).collect(Collectors.toMap(Pair::getKey, Pair::getValue)));
+      Pair.of(entry.getValue().getFileGroupId(), Pair.of(entry.getKey(), entry.getValue()))).collect(Collectors.toConcurrentMap(Pair::getKey, Pair::getValue)));
```

Comment:
nit: in `HoodieTableFIleSystemView`, we have methods like `createFileIdToBootstrapBaseFileMap`, `createFileIdToReplaceInstantMap`, etc.  Some existing methods including `createFileIdToReplaceInstantMap` have the logic of instantiating concurrent hash map there.   Should we just override the implementation of `createFileIdToBootstrapBaseFileMap` instead of changing the logic here?